### PR TITLE
Performance

### PR DESCRIPTION
**Pull request by thiago-cavalcante**
@lucasccordeiro , could you please review this PR?

### CODE REVIEW

File: src/dsverifier.cpp [hunk 1]
```diff
@@ -156,6 +157,10 @@ double desired_quantization_limit = 0.0;
 bool show_counterexample_data = false;
 bool preprocess = false;
 
+/* MACROS */
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Can you add some description about these macros?

File: src/dsverifier.cpp [hunk 2]
```diff
@@ -1987,7 +1991,7 @@ bool isEigPos(Eigen::MatrixXd A)
   for(i = 0; i < _controller.nStates; i++)
   {
     lambda = eivals[i];
-    if(lambda.real() >= 0)
+    if((lambda.real() >= 0) && (lambda.imag() == 0))
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Replace ``lambda.imag() == 0`` by ``!lambda.imag()``.

File: src/dsverifier.cpp [hunk 3]
```diff
@@ -2016,45 +2021,58 @@ bool isEigPos(Eigen::MatrixXd A)
...
-    cur = y_k(A, B, C, D, u, i+1, x0);
-    pos = y_k(A, B, C, D, u, i+2, x0);
-    out[1] = pre;
-    out[0] = i;
-    peak = pre;
-    while((fabs(out[1]) <= fabs(peak)))
+  int i = 0, numBadPeaks = 0, firstGradSampleIdx, lastPeakIdx, lastGrad = 1, grad;
+  double lastPeak, firstGradSample;
+  lastPeak = y_k(A, B, C, D, u, i, x0);
+    while(1)
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp:2032]: incorrect indentation.

File: src/dsverifier.cpp [hunk 4]
```diff
@@ -2016,45 +2021,58 @@ bool isEigPos(Eigen::MatrixXd A)
...
     {
-      if((out[1] != cur))
+      if(fabs(y_k(A, B, C, D, u, i+1, x0)) >= fabs(y_k(A, B, C, D, u, i, x0)))
+      {
+        grad = (grad > 0)?(grad + 1):1;
+        if(fabs(y_k(A, B, C, D, u, i+1, x0)) != fabs(y_k(A, B, C, D, u, i, x0)))
+        {
+  	      firstGradSample = y_k(A, B, C, D, u, i+1, x0);
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Incorrect indentation. Please run https://github.com/ssvlab/dsverifier/blob/master/scripts/cpplint.py

File: src/dsverifier.cpp [hunk 5]
```diff
@@ -2016,45 +2021,58 @@ bool isEigPos(Eigen::MatrixXd A)
...
+        }
+      }
+      else
+      {
+        grad = (grad < 0)?(grad - 1):-1;
+      }
+      if((lastGrad > 0) && (grad < 0))
       {
-        if((fabs(cur) >= fabs(pos)) && (fabs(cur) >= fabs(pre)))
+        if(fabs(firstGradSample) <= fabs(lastPeak))
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp:2049]: Incorrect indentation. Please run https://github.com/ssvlab/dsverifier/blob/master/scripts/cpplint.py

File: src/dsverifier.cpp [hunk 6]
```diff
@@ -2016,45 +2021,58 @@ bool isEigPos(Eigen::MatrixXd A)
...
+            break;
+  	      }
         }
-        if((out[1] != peak) && (isSameSign(yss, peak)) &&
-           (fabs(peak) > fabs(out[1])))
+        else
         {
-          out[0] = i+1;
-          out[1] = peak;
+  	      lastPeak = firstGradSample;
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Incorrect indentation. Please run https://github.com/ssvlab/dsverifier/blob/master/scripts/cpplint.py

File: src/dsverifier.cpp [hunk 7]
```diff
@@ -2179,24 +2197,26 @@ int check_settling_time(Eigen::MatrixXd A, Eigen::MatrixXd B,
                         Eigen::MatrixXd x0, double u, double tsr,
                         double p, double ts)
 {
-  double peakV[2];
-  double yss, yp, lambMax, cbar, output, inf, sup, v;
+//  double peakV[2];
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Remove this.

File: src/dsverifier.cpp [hunk 8]
```diff
@@ -2208,13 +2228,28 @@ int check_settling_time(Eigen::MatrixXd A, Eigen::MatrixXd B,
...
+    if(fabs((fabs(yp) - fabs(yss))) <= 0.00001)
+    {
+      v = y_k(A, B, C, D, u, i, x0);
+      while(!((v < sup) && (v > infe)))
+      {
+        i++;
+   	    v = y_k(A, B, C, D, u, i, x0);
+   	  }
+   	  kbar = i+1;
+      std::cout << "khat=" << kbar << std::endl;
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp:2247]: Incorrect indentation. Please run https://github.com/ssvlab/dsverifier/blob/master/scripts/cpplint.py

File: src/dsverifier.cpp [hunk 9]
```diff
@@ -2347,13 +2381,16 @@ int check_overshoot(Eigen::MatrixXd A, Eigen::MatrixXd B,
                     Eigen::MatrixXd C, Eigen::MatrixXd D,
                     Eigen::MatrixXd x0, double u, double _POr)
 {
-  double peakV[2];
+//  double peakV[2];
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: Remove this.

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
@thiago-cavalcante: Can you please:

(1) Correct code indentation. 
(2) Run cpplint in your code https://github.com/ssvlab/dsverifier/blob/master/scripts/cpplint.py
(3) Add some tests to our regression suite related to performance requirements.

File: src/dsverifier.cpp (hunk 1 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: DONE

File: src/dsverifier.cpp (hunk 2 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: DONE

File: src/dsverifier.cpp (hunk 4 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: DONE

File: src/dsverifier.cpp (hunk 3 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp:2032]: DONE

File: src/dsverifier.cpp (hunk 5 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp:2049]: DONE

File: src/dsverifier.cpp (hunk 6 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: DONE

File: src/dsverifier.cpp (hunk 7 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: DONE

File: src/dsverifier.cpp (hunk 8 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp:2247]: DONE

File: src/dsverifier.cpp (hunk 9 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: DONE

File: src/dsverifier.cpp [hunk 10]
```diff
@@ -1991,7 +1995,11 @@ bool isEigPos(Eigen::MatrixXd A)
   for(i = 0; i < _controller.nStates; i++)
   {
     lambda = eivals[i];
+<<<<<<< HEAD
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: What does ``<<<<<<< HEAD'' mean?

File: src/dsverifier.cpp [hunk 11]
```diff
@@ -1991,7 +1995,11 @@ bool isEigPos(Eigen::MatrixXd A)
...
+    if((lambda.real() >= 0) && (!lambda.imag()))
+=======
```
Comment (lucasccordeiro) [inline src/dsverifier.cpp]: What does "=======" mean?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
There are many ``<<<<<<< HEAD`` and ``=======`` throughout the code.

Can you also please squash the commits and add meaningful commit messages?

File: src/dsverifier.cpp (hunk 10 above)
Comment (thiago-cavalcante) [reply src/dsverifier.cpp]: It was already fixed in the las commit.

**Review by lucasccordeiro — APPROVED** [review]
Looks good to me.

**Comment by lucasccordeiro** [comment]
@lennonchaves: May I ask you to review this PR?

**Review by lennonchaves — APPROVED** [review]

**Comment by lennonchaves** [comment]
Hi @thiago-cavalcante , please go ahead to merge the pull request thanks.

**Pull request merged**